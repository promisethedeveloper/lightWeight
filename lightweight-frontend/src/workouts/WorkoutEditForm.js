import React, { useState } from "react";
import { useHistory, useParams } from "react-router-dom";
import Alert from "../common/Alert";
import LightWeightAPI from "../api/api";

const WorkoutEditForm = () => {
	const { day, description } = useParams();

	const INITIALSTATE = {
		dayOfWeek: day,
		description: description,
	};

	// const { currentUser } = useContext(UserContext);
	// const { id: user_id } = currentUser;

	const history = useHistory();
	const [formData, setFormData] = useState(INITIALSTATE);

	const [formErrors, setFormErrors] = useState([]);

	/**
	 * Handle form submit:
	 *
	 * Calls login function prop and if successful, redirect to "/workouts"
	 */

	const handleSubmit = async (e) => {
		e.preventDefault();
		try {
			await LightWeightAPI.editWorkout(formData.dayOfWeek, formData);
			history.push("/workouts");
			setFormData(INITIALSTATE);
		} catch (error) {
			setFormErrors(error);
		}
	};

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((data) => ({ ...data, [name]: value }));
	};

	return (
		<div className="LoginForm">
			<div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
				<h3 className="mb-3">Edit workout day</h3>

				<div className="card">
					<div className="card-body">
						<form onSubmit={handleSubmit}>
							<div className="form-group">
								<label htmlFor="dayOfWeek">Day of week</label>
								<input
									id="dayOfWeek"
									name="dayOfWeek"
									className="form-control"
									value={formData.dayOfWeek}
									onChange={handleChange}
									autoComplete="username"
									placeholder="ex. Monday"
									required
								/>
							</div>
							<div className="form-group">
								<label htmlFor="description">Description</label>
								<input
									id="description"
									name="description"
									className="form-control"
									value={formData.description}
									onChange={handleChange}
									placeholder="ex. Chest day"
									required
								/>
							</div>
							{formErrors.length ? (
								<Alert type="danger" messages={formErrors} />
							) : null}
							<button
								className="btn float-right font-weight-bold"
								style={{ backgroundColor: "#461111", color: "white" }}
							>
								Update
							</button>
							<a
								className="btn float-right font-weight-bold"
								style={{
									backgroundColor: "#FF1E00",
									color: "white",
									marginRight: "5px",
								}}
								href="/workouts"
								role="button"
							>
								Cancel
							</a>
						</form>
					</div>
				</div>
			</div>
		</div>
	);
};

export default WorkoutEditForm;
