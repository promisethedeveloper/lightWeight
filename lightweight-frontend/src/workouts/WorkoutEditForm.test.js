import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import WorkoutEditForm from "./WorkoutEditForm";

/** Smoke test */
it("renders without crashing", () => {
	render(
		<MemoryRouter>
			<WorkoutEditForm />
		</MemoryRouter>
	);
});

/** Snapshot test */
it("Should match snapshots", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<WorkoutEditForm />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("renders a 'Edit workout day text", () => {
	const { getByText } = render(
		<MemoryRouter>
			<WorkoutEditForm />
		</MemoryRouter>
	);
	expect(getByText("Edit workout day")).toBeInTheDocument();
});

it("renders form's label text", () => {
	const { getByLabelText } = render(
		<MemoryRouter>
			<WorkoutEditForm />
		</MemoryRouter>
	);
	expect(getByLabelText("Day of week")).toBeInTheDocument();
	expect(getByLabelText("Description")).toBeInTheDocument();
});

it("renders form's Update text", () => {
	const { getByText } = render(
		<MemoryRouter>
			<WorkoutEditForm />
		</MemoryRouter>
	);
	expect(getByText("Update")).toBeInTheDocument();
});
