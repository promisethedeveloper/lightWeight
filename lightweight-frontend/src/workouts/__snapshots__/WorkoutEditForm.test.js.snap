// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshots 1`] = `
<DocumentFragment>
  <div
    class="LoginForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h3
        class="mb-3"
      >
        Edit workout day
      </h3>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                for="dayOfWeek"
              >
                Day of week
              </label>
              <input
                autocomplete="username"
                class="form-control"
                id="dayOfWeek"
                name="dayOfWeek"
                placeholder="ex. Monday"
                required=""
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="description"
              >
                Description
              </label>
              <input
                class="form-control"
                id="description"
                name="description"
                placeholder="ex. Chest day"
                required=""
                value=""
              />
            </div>
            <button
              class="btn float-right font-weight-bold"
              style="background-color: rgb(70, 17, 17); color: white;"
            >
              Update
            </button>
            <a
              class="btn float-right font-weight-bold"
              href="/workouts"
              role="button"
              style="background-color: rgb(255, 30, 0); color: white; margin-right: 5px;"
            >
              Cancel
            </a>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
